fedimg/utils.py

```diff
@@ -88,16 +89,67 @@ def ssh_connection_works(username, ip, keypath):
         works = True
     except (paramiko.BadHostKeyException,
             paramiko.AuthenticationException,
-            paramiko.SSHException, socket.error) as e:
+            paramiko.SSHException, socket.error):
         pass
     ssh.close()
     return works
 
 
-def safeget(dct, *keys):
+def get_value_from_dict(_dict, *keys):
     for key in keys:
         try:
-            dct = dct[key]
+            _dict = _dict[key]
         except KeyError:
             return None
-    return dct
+    return _dict
+
+
+def external_run_command(command):
+    ret = subprocess.Popen(' '.join(command), shell=True, stdin=subprocess.PIPE,
+                           stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                           close_fds=True)
+    out, err = ret.communicate()
+    retcode = ret.returncode
+    return out, err, retcode
+
+
+def get_item_from_regex(output, regex):
+    match = re.search(regex, output)
+    if match is None:
+        return ''
+    else:
+        return match.group(1)
+
+
+def get_file_name_image(image_url):
+    return image_url.split('/')[-1]
+
+
+def get_source_from_image(image_url):
+    tmpdir = tempfile.mkdtemp()
+    file_name = get_file_name_image(image_url)
+    file_path = os.path.join(tmpdir, file_name)
+
+    log.info("[PREP] Preparing temporary directory for download: %r" % tmpdir)
+    output, error, retcode = external_run_command([
+        'wget',
+        image_url,
+        '-P',
+        tmpdir
+    ])
```

Comment:
You're not checking the returned code to know if the command succeeded or not, maybe you could do that in ``external_run_command()`` directly?

Reply:
Yeah! this downloads the image. I should surely check and stop the process if the file is not downloaded :100:

Reply:
Still applies :)

Reply:
Fixed this.